src/components/mapPageComponents/mapView/index.tsx

```diff
@@ -61,6 +63,13 @@ const MapView: React.FC<MapViewProps> = ({
   );
```

Comment:
❗ Wow there are over 400 lines of code in this react component excluding JSX. That is _way_ too much. Almost all of this should be extracted into another file.

Reply:
Agreed, I'll open a ticket for cleanup